File: app/src/main/java/com/example/whattowatch/model/mappers/Detail2ListMapper.java
package com.example.whattowatch.model.mappers;

import android.util.Log;

import com.example.whattowatch.model.mymodel.MyDetailModel;
import com.example.whattowatch.model.mymodel.MyMovieModel;

import java.util.ArrayList;
import java.util.List;

public class Detail2ListMapper {

    public static List<MyMovieModel> convert2ListModel(List<MyDetailModel> detailModelList){
        List<MyMovieModel> list = new ArrayList<>();

        for(MyDetailModel detailModel: detailModelList){
           MyMovieModel model = new MyMovieModel(detailModel.getName(),
                   detailModel.getOverview(),
                   detailModel.getReleaseDate(),
                   detailModel.getRate(),
                   detailModel.getPosterLink(),
                   detailModel.getId(),
                   "favorite");
            list.add(model);
            Log.e("mylog","Detail2ListMapper");
        }
        return list;
    }
}

File: app/src/main/java/com/example/whattowatch/model/mappers/Detail2ListMapper.kt
package com.example.whattowatch.model.mappers

import android.util.Log
import com.example.whattowatch.model.mymodel.MyDetailModel
import com.example.whattowatch.model.mymodel.MyMovieModel
import java.util.*

object Detail2ListMapper {
    fun convert2ListModel(detailModelList: List<MyDetailModel>): List<MyMovieModel> {
        val list: MutableList<MyMovieModel> = ArrayList()
        for (detailModel in detailModelList) {
            val model = MyMovieModel(detailModel.name,
                    detailModel.overview,
                    detailModel.releaseDate,
                    detailModel.rate,
                    detailModel.posterLink,
                    detailModel.id,
                    "favorite")
            list.add(model)
            Log.e("mylog", "Detail2ListMapper")
        }
        return list
    }
}